src/QsCompiler/Transformations/Monomorphization.cs

```diff
@@ -87,13 +225,91 @@ public override TypedExpression Transform(TypedExpression ex)
             return base.Transform(ex);
         }
 
-        private string CreateTypePrefix(ImmutableDictionary<QsTypeParameter, ResolvedType> typeDict)
+        private Concretion CreateConcretion(ImmutableDictionary<QsTypeParameter, ResolvedType> typeDict)
         {
-            return "_" + String.Join("", typeDict
-                .ToList()
-                .OrderBy(kvp => kvp.Key.TypeName)
-                .Select(kvp => kvp.Key.TypeName.Value + "_" + kvp.Value.Resolution.ToString() + "_")
-                );
+            Concretion result = new Concretion();
+            foreach (var kvp in typeDict)
+            {
+                result.Add((kvp.Key, kvp.Value));
+            }
+            return result;
+        }
+
+        private Identifier GetConcreteIdentifier(Identifier.GlobalCallable globalCallable, ImmutableDictionary<QsTypeParameter, ResolvedType> types)
+        {
+            Concretion target = null;
```

Comment:
I think SetEquals below might actually throw if target is null.

Reply:
This issue is addressed in PR #206.